src/App.tsx

```diff
@@ -1,24 +1,227 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import {
+  addTodos,
+  deleteTodos,
+  getTodos,
+  patchTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+const USER_ID = 27;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [error, setError] = useState<Error | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setError(Error.UnableToLoadAll));
+  }, []);
+
+  const filterTodos = useCallback((currentTodos: Todo[], query: Filter) => {
+    return currentTodos.filter(todo => {
+      switch (query) {
+        case Filter.All:
+          return todo;
+        case Filter.Completed:
+          return todo.completed;
+        case Filter.Active:
+          return !todo.completed;
+        default:
+          return todo;
+      }
```

Comment:
The first case is redundant, because you return todo in default
```suggestion
      switch (query) {
        case Filter.Completed:
          return todo.completed;
          
        case Filter.Active:
          return !todo.completed;
          
        default:
          return todo;
      }
```